### ISSUE TITLE
"Using topology(a_graph) { ... distance_to...}" use the perimeter of the edges rather their weight

### ISSUE THREAD

**Issue by ptaillandier** [gama-platform/gama#3273]
**Describe the bug**
When using the structure using topology(a_graph), GAMA uses the perimeter of the edges to compute distances rather than the weight of the edges.

**To Reproduce**
test the following model:
```
model initialization

global {
	  
  graph road_network; 
  
  init {
  		create road  with:(shape:line([{10,10}, {90,90}]));
  		
		road_network <- as_edge_graph(road) with_weights ([road[0]::1.0]);
		using topology(road_network) {
        	write sample({10,10} distance_to {90,90});
         }
         write "good result:" + sample((road_network path_between ({10,10},{90,90} )).weight);
	}
}

species road ;

experiment main type: gui ;
```

**Comment by NaroaCS** [comment]
Hi @ptaillandier ! I think I am facing an issue similar to this one. Please, let me know if it's fine to post this here or if I should create a new issue. Thanks in advance for any help !

I have been facing a strange behavior when using the function `distance_to using topology(roadNetwork)`. (See code [here](https://github.com/NaroaCS/MultifunctionalVehicles/blob/e36d68a5e3240894e87de14c9a8d23bb3b5d1c64/models/Agents.gaml#L9) ). 

If I use the regular `distance_to`, the distances are in the order of 20-1000m, which seems quite right for my project. But if I use `distance_to using topology(roadNetwork)` it gives distances in the orders of 20,000m and it doesn't make any sense. For the rest, the roadNetwork is working well, so I don't think the roadNetwork is the problem.

This did not happen when I was using GAMA 1.8.1 but it started to happen when using the 1.8.2 releases (currently, I am using [Alpha Version 1.8.2 (0148ff0b)](https://github.com/gama-platform/gama/releases/tag/1.8.2)).

**Comment by AlexisDrogoul** [comment]
Hi @NaroaCS ,

I would tend to think that it could be better to open a new issue if you do not mind, together with a small test model. 

Thanks in advance ! 
Alexis

**Comment by NaroaCS** [comment]
Hi @AlexisDrogoul! Thanks for the quick reply. 
Trying to replicate the error in a toy model, I realized that the problem was that my people agents were in buildings, so in order to use `distance_to using topology(roadNetwork)` I needed to use the vertice closest to the agent (`roadNetwork.vertices closest_to(person)`).  This totally makes sense, so I think it was just that mistake on my side; sorry for the trouble!
In case this is not the expected behavior I'm happy to share the toy model too.
